functionary/builder/utils.py

```diff
@@ -105,6 +255,100 @@ def initiate_build(
     return build
 
 
+def create_package_from_definition(
+    package_definition: dict, environment: "Environment"
+) -> Package:
+    """Create a package from a package definition"""
+    # TODO: Manually parsing for now, but this should be codified somewhere, with
+    #       the parsing informed by package_definition_version.
+    package_obj = Package(
+        environment=environment,
+        name=package_definition.get("name"),
+    )
+
+    package_obj.display_name = package_definition.get("display_name")
+    package_obj.summary = package_definition.get("summary")
+    package_obj.description = package_definition.get("description")
+    package_obj.language = package_definition.get("language")
+
+    package_obj.save()
+
+    return package_obj
+
+
+@app.task
+def build_package(build_id: "UUID") -> None:
+    """Retrieve the resources for Build and use them to build and push the package
+    docker image. Also creates BuildLog with build/push information.
+
+    Args:
+        build_id: ID of the build being executed
+
+    Returns:
+        None
+    """
+    build = Build.objects.select_related("environment").get(id=build_id)
+    build_resource = BuildResource.objects.get(build=build)
+    log_msgs: list[str] = []
+
+    logger.info(f"Starting build {build.id}")
+
+    try:
+        _build_package(build, build_resource, log_msgs)
+        build.complete()
+    except Exception as err:
+        logger.error(f"Build {build.id} encountered error: {err}")
+        log_msgs.append(str(err))
+        build.error()
+    finally:
+        log = "\n".join(log_msgs)
+        _create_build_log(build, log)
+
+
+def _build_package(build: Build, build_resource: BuildResource, log: list[str]) -> None:
+    """Generate and build necessary resources to complete image build
+
+    Args:
```

Comment:
The docstring doesn't accurately reflect the function bahavior after recent changes.  Since this is an internal helper function, you can just have the one line summary docstring if you want.  Otherwise, let's update this to reflect the current state of things.